File: 12.integer-to-roman.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution12
{
    public string IntToRoman(int num) 
    {
        String[] M = new string[] { "", "M", "MM", "MMM" };
        String[] C = new string[] { "", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM" };
        String[] X = new string[] { "", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC" };
        String[] I = new string[] { "", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX" };
        
        return M[num / 1000] + C[(num % 1000) / 100] + X[(num % 100) / 10] + I[num % 10];
    }
}

File: 12_integer-to-roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        roman=''
        roman_dict = {1000 : 'M',
                    900 : 'CM',
                    500 : 'D',
                    400 : 'CD',
                    100 : 'C',
                    90 : 'XC',
                    50 : 'L',
                    40 : 'XL',
                    10 : 'X',
                    9 : 'IX',
                    5 : 'V',
                    4 : 'IV',
                    1 : 'I'}
        for i in roman_dict:
            roman += roman_dict[i]*(num // i)
            num = num % i
            if num == 0:
                break
        return roman

instance = Solution()
print(instance.intToRoman(8))